### ISSUE TITLE
Handle same nicknames

### ISSUE THREAD

**Issue by Sorunome** [Half-Shot/matrix-appservice-discord#334]
handle somehow multiple people on matrix having the same nicknames on discord, e.g. embeds with server name or something

**Comment by schrmh** [comment]
Why this is a problem:
![image](https://user-images.githubusercontent.com/32062480/49701249-38e12b80-fbea-11e8-854a-53f599470103.png)
comes out as
![image](https://user-images.githubusercontent.com/32062480/49701254-4b5b6500-fbea-11e8-8caa-c5a2b83a0223.png)
on Discord side.
So if there are multiple users with the same name on matrix you can't see on Discord side which one wrote. It treats all of them as one user and even combines their message like in the screenshot if two different users with the same name wrote after another on matrix.

**What can Discord webhook do:**
- Icons
- Have names up to 32 chars.
- Set embed/message properties (embeds doesn't count to the 2000 char limit?)

What can't Discord do (but would be the best solution imo):
- Colored names
- Roles for webhook

This leads us to following workarounds:
**1) Include domain name in webhook display name**
But well 32 chars (https://discordapp.com/developers/docs/resources/webhook#create-webhook) are pretty limited..

**2) Use different icons for different users**
Well as you can see on the screenshot both of my accounts use the default icon. May be possible to distinguish between them by e.g. using the icon of the website where the account is registered at or something but I consider this as somewhat unpractical.

**3) Change things in messages.** 
Seems to be the way to go at the moment.
**3.1)** Add domain in raw format to the message. May be the easiest thing but definitely counts to the 2000 char limit.
**3.2)** Put entire messages in an (colored) embed (with domain as title). Considered as ugly.
**3.3)** Send two messages if multiple users with the same name are on a channel.
So the first one would be a embed which has no content and just a title showing the domain and the second one would be the actual message that was sent by the user on matrix.


Those are my ideas so far; maybe you come up with something else or you know something important (e.g. that webhooks can do more than what I wrote).

**Comment by Sorunome** [comment]
soru would be leaning against 3.3, only to trigger if on the matrix side are two users of the same name, though

**Comment by pacien** [comment]
Yes please. Discord's current bad handling of profile pictures for webhook messages is adding even more confusion (icons are sometimes swapped).
